File: chaincode/modules/part.go
package modules

// Part 零部件核心信息
type Part struct {
	PartID       string `json:"partID"`       // 零部件唯一标识（如"ENG-PISTON-202505"）
	VIN          string `json:"vin"`          // 关联车辆VIN码（如有）
	BatchNo      string `json:"batchNo"`      // 生产批次号
	Name         string `json:"name"`         // 零部件名称（如"发动机活塞"）
	Type         string `json:"type"`         // 类型（如"发动机部件"）
	Manufacturer string `json:"manufacturer"` // 生产厂商ID（关联身份管理）
	CreateTime   string `json:"createTime"`   // 创建时间戳（上链时间）
	Status       string `json:"status"`       // 状态（在产/在售/召回/报废等）
}

// PartLifecycle 零部件全生命周期轨迹
type PartLifecycle struct {
	PartID          string            `json:"partID"`          // 关联零部件
	BOMInfo         BOMReference      `json:"bomInfo"`         // 关联BOM信息
	ProductionInfo  ProductionData    `json:"productionInfo"`  // 关联生产数据
	QualityInfo     QualityInspection `json:"qualityInfo"`     // 关联质检数据
	SupplyChainInfo []SupplyChainData `json:"supplyChainInfo"` // 关联供应链数据（订单+物流）
	AftersaleInfo   []AftersaleRecord `json:"aftersaleInfo"`   // 关联售后/故障数据
}

// BOMReference BOM信息引用
type BOMReference struct {
	BOMID   string `json:"bomID"`   // 关联BOM的ID
	Version string `json:"version"` // 关联BOM版本
	Type    string `json:"type"`    // 研发BOM/生产BOM
}

File: frontend/src/types/part.ts
/**
 * 零部件相关类型定义
 * 定义零部件管理模块的数据类型和接口
 */

// 零部件信息
export interface Part {
  partID: string        // 零部件唯一标识
  vin: string           // 关联车辆VIN码
  batchNo: string       // 生产批次号
  name: string          // 零部件名称
  type: string          // 类型
  manufacturer: string  // 生产厂商ID
  createTime: string    // 创建时间戳
  status: string        // 状态（在产/在售/召回/报废等）
}

// 零部件生命周期
export interface PartLifecycle {
  partID: string
  bomInfo?: BOMReference
  productionInfo?: ProductionData
  qualityInfo?: QualityInspection
  supplyChainInfo?: SupplyChainData[]
  aftersaleInfo?: AftersaleRecord[]
}

// BOM引用信息
export interface BOMReference {
  bomID: string
  version: string
  type: string
}

// 生产数据
export interface ProductionData {
  productionID: string
  partID: string
  batchNo: string
  params: Record<string, string>
  productionLine: string
  operator: string
  finishTime: string
}

// 质检数据
export interface QualityInspection {
  inspectionID: string
  partID: string
  batchNo: string
  indicators: Record<string, string>
  result: string
  handler: string
  handleTime: string
  disposal: string
}

// 供应链数据
export interface SupplyChainData {
  chainID: string
  partID: string
  batchNo: string
  orderID: string
  logisticsID: string
  stageType: string
  stageStatus: string
  participator: string
  participatorRole: string
  quantity: number
  operateTime: string
  operator: string
  remark: string
}

// 售后记录
export interface AftersaleRecord {
  aftersaleID: string
  partID: string
  batchNo: string
  vin: string
  faultID: string
  recallID: string
  aftersaleType: string
  aftersaleStatus: string
  handlerOrgID: string
  handlerID: string
  description: string
  processResult: string
  processTime: string
  cost: string
  remark: string
}

// 创建零部件请求
export interface CreatePartRequest {
  partID: string
  vin?: string
  batchNo: string
  name: string
  type: string
  manufacturer: string
  status?: string
}

// 更新零部件状态请求
export interface UpdatePartStatusRequest {
  partID: string
  status: string
}
